### ISSUE TITLE
Q: Best practice on storing access tokens / refresh tokens

### ISSUE THREAD

**Issue by moeffju** [oblador/react-native-keychain#387]
We use OAuth to authenticate clients, including our mobile app, so it gets an `access_token` and `refresh_token`. Is there a best practice / suggestion on what data to store with react-native-keychain for this case, and how? i.e. should we use generic password and add two separate entries, should we store the username & password and put the access token somewhere else..? Any hints are appreciated.

**Comment by Iltimirov** [comment]
as an option you can use this approach https://github.com/oblador/react-native-keychain/issues/291#issuecomment-682460091

**Comment by moeffju** [comment]
Thank you for the pointer!

**Comment by acatalina** [comment]
I wouldn't store username and password, that's what access and refresh tokens are there for, to represent the credentials of a user. Also, tokens are revokable whereas passwords are not so security wise wouldn't makes sense to store the user's credentials neither.

**Comment by moeffju** [comment]
@acatalina Yeah, that’s why I’m asking about the best practice to store these tokens :) The public APIs of r-n-k are all designed for usernames & passwords, thus my question about what the suggested way is to use them to store the tokens instead :)

**Comment by acatalina** [comment]
oh, fair enough hehe Yes, the only thing I'd say (following @Iltimirov recommendation) is to namespace the service key to your application so you don't get into any clashes

**Comment by DorianMazur** [comment]
This issue is stale because it has been open for 1 year with no activity.
